src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -347,21 +347,35 @@ internal IOperation CreateInternal(BoundNode boundNode)
                         }
                     }
 
-                    return new CSharpLazyNoneOperation(this, boundNode, _semanticModel, boundNode.Syntax, constantValue, isImplicit: isImplicit, type: null);
+                    ImmutableArray<IOperation> children = GetIOperationChildren(boundNode);
+                    return new NoneOperation(children, _semanticModel, boundNode.Syntax, type: null, constantValue, isImplicit: isImplicit);
 
                 default:
                     throw ExceptionUtilities.UnexpectedValue(boundNode.Kind);
             }
         }
 
+#nullable enable
         private IMethodBodyOperation CreateMethodBodyOperation(BoundNonConstructorMethodBody boundNode)
         {
-            return new CSharpLazyMethodBodyOperation(this, boundNode, _semanticModel, boundNode.Syntax);
+            return new MethodBodyOperation(
+                (IBlockOperation?)Create(boundNode.BlockBody),
+                (IBlockOperation?)Create(boundNode.ExpressionBody),
+                 _semanticModel,
+                 boundNode.Syntax,
+                 isImplicit: boundNode.WasCompilerGenerated);
         }
 
         private IConstructorBodyOperation CreateConstructorBodyOperation(BoundConstructorMethodBody boundNode)
         {
-            return new CSharpLazyConstructorBodyOperation(this, boundNode, boundNode.Locals.GetPublicSymbols(), _semanticModel, boundNode.Syntax);
+            return new ConstructorBodyOperation(
+                boundNode.Locals.GetPublicSymbols(),
+                Create(boundNode.Initializer),
+                (IBlockOperation?)Create(boundNode.BlockBody),
+                (IBlockOperation?)Create(boundNode.ExpressionBody),
+                _semanticModel,
+                boundNode.Syntax,
+                isImplicit: boundNode.WasCompilerGenerated);
         }
 
         internal ImmutableArray<IOperation> GetIOperationChildren(BoundNode boundNode)
```

Comment:
The `boundNode` argument is only used as `IBoundNodeWithIOperationChildren`. Why don't we make that the parameter type of this method?

Reply:
We could, but I didn't want to make changes to code where I could avoid it (to minimize potential for changes). I can do this in a follow up.